ru/lessons/specifics/mnesia.md

```diff
@@ -174,3 +181,119 @@ iex> data_to_read = fn ->
 ```
+
+Для редактирования достаточно вызвать `Mnesia.write/1` с ключом уже существующей записи. Например, обновление записи `Hans` выглядело бы так:
+
+```elixir
+iex> Mnesia.transaction(
+...>   fn ->
+...>     Mnesia.write({Person, 5, "Hans Moleman", "Ex-Mayor"})
+...>   end
+...> )
+```
+
+## Использование индексов
+
+Mnesia поддерживает создание индексов на неключевых колонках и позволяет запрашивать данные на их основе. Например, можно добавить индекс для колонки `:job` таблицы `Person`:
+
+```elixir
+iex> Mnesia.add_table_index(Person, :job)
+{:atomic, :ok}
+```
+
+Возвращаемый результат идентичен результату функции `Mnesia.create_table/2`:
+
+ - `{:atomic, :ok}`, если функция выполнилась успешно;
+ - `{:aborted, Reason}` в случае ошибки.
+
+В частности, если индекс уже существует, ответ будет в виде `{:already_exists, table, attribute_index}`. Таким образом, если мы попробуем пересоздать индекс, то получим следующее:
+
+```elixir
+iex> Mnesia.add_table_index(Person, :job)
+{:aborted, {:already_exists, Person, 4}}
+```
+
+После успешного создания индекса можно использовать его для чтения данных. Например, получим список директоров:
+
+```elixir
+iex> Mnesia.transaction(
+...>   fn ->
+...>     Mnesia.index_read(Person, "Principal", :job)
+...>   end
+...> )
+{:atomic, [{Person, 1, "Seymour Skinner", "Principal"}]}
+```
+
+## Match и select
+
+Mnesia поддерживает сложные запросы данных из таблиц в виде сопоставления и функций для выборки.
+
+Функция `Mnesia.match_object/1` возвращает все записи, соответствующие образцу. Если на колонках существуют индексы, она может использовать их для повышения эффективности запроса. Для колонок, не участвующих в сопоставлении используется `:_`.
+
+```elixir
+iex> Mnesia.transaction(
+...>   fn ->
+...>     Mnesia.match_object({Person, :_, "Marge Simpson", :_})
+...>   end
+...> )
+{:atomic, [{Person, 4, "Marge Simpson", "home maker"}]}
+```
+
+Функция `Mnesia.select/2` позволяет составлять запросы с использованием операторов и функций Elixir или Erlang. Например, выберем записи, у которых ключ больше 3:
+
+```elixir
+iex> Mnesia.transaction(
+...>   fn ->
+...>     {% raw %}Mnesia.select(Person, [{{Person, :"$1", :"$2", :"$3"}, [{:>, :"$1", 3}], [:"$$"]}]){% endraw %}
+...>   end
+...> )
+{:atomic, [[7, "Waylon Smithers", "Executive assistant"], [4, "Marge Simpson", "home maker"], [6, "Monty Burns", "Businessman"], [5, "Hans Moleman", "unknown"]]}
+```
+
+Разберём по частям. Первый атрибут &mdash; таблица `Person`, второй &mdash; кортеж вида `{match, [guard], [result]}`:
+
+- `match` &mdash; то же самое, что передаётся в функцию `Mnesia.match_object/1`. Обратите внимание на специальные атомы вида `:"$n"`, используемые как позиционные параметры для колонок.
+- `guard` &mdash; список кортежей с описанием функций-ограничителей. В нашем случае это встроенная функция `:>` (знак "больше") с первым позиционным параметром `:"$1"` и константой `3` в качестве атрибутов.
```

Comment:
I think the punctuation here is already right. 
http://new.gramota.ru/spravka/punctum?layout=item&id=58_126